### PR TITLE
Overhaul of AngularJS Module

### PR DESCRIPTION
**Pull request by coryasilva**
Refactored a bunch of nonsense code and made the HTML report a lot prettier.  Now we are properly using the REST endpoint as a POST, no longer restricted by the URL lengths.

### CODE REVIEW

File: client/bugLogClientAngular.js [hunk 1]
```diff
@@ -1,134 +1,171 @@
...
-    }
-])
-.factory('BugLogService', ["$window", "$document", "appName", "bugLogConfig",
-    function BugLogService($window, $document, appName, bugLogConfig) {
-        var BugLog = {
+    //the application to tell BugLog from where the report is coming from.
+    .constant('appName', "app")
+    .constant('bugLogConfig', {
+        // this is the location of the BugLog server that will be receiving the error reports (needs to be a REST/POST endpoint)
+        listener: "",
```
Comment (oarevalo) [inline client/bugLogClientAngular.js]: I noticed that the option to override the HostName is removed. I know there are multiple cases in which users prefer to set the hostname explicitly. If not present, BugLogHQ will revert to use the CGI information, which may or may not be what you want.

File: client/bugLogClientAngular.js [hunk 2]
```diff
@@ -1,134 +1,171 @@
...
+         * @param {string} cause
+         * @param {object} error
+         * @param {string} severityCode
+         */
+        function logBug(cause, error, severityCode) {
+            var data = {};
+            data.message = 'Caused by: ' + cause;
+            data.exceptionMessage = cause || 'Unknown';
+            data.exceptionDetails =  buildStacktrace(error) || 'something went wrong';
+            data.severityCode = severityCode || 'EXCEPTION';
```
Comment (oarevalo) [inline client/bugLogClientAngular.js]: Have you thought about making the default severity code a constant or something that can be configured somewhere? Alternatively having a consistent default value between client and server would also make sense. In the server, the default when severityCode is not provided is `ERROR`

File: client/bugLogClientAngular.js (hunk 1 above)
Comment (coryasilva) [inline client/bugLogClientAngular.js]: Yeah that makes sense; I will add it back.

File: client/bugLogClientAngular.js (hunk 2 above)
Comment (coryasilva) [inline client/bugLogClientAngular.js]: Easily

**Pull request closed without merging**